# Module 3 Lab 2: Outliers in the Crime Data


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("usarrests.csv")


In [3]:
df.head()

,Unnamed: 0,Murder,Assault,UrbanPop
0,Alabama,13.2,236.0,58
1,Alaska,10.0,263.0,48
2,Arizona,8.1,294.0,80
3,Arkansas,8.8,190.0,50
4,California,9.0,276.0,91


## Summary Statistics

In [4]:
df.describe()

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


## Investigating Suspicious Values


In [5]:
df[(df["Assault"].isna()) | (df["Assault"] > 500) | (df["UrbanPop"] > 100)]


,Unnamed: 0,Murder,Assault,UrbanPop
9,Georgia,17.4,NaN,60
14,Iowa,2.2,56.0,570
39,South Carolina,14.4,879.0,48


In [6]:
df[(df["Murder"] < 0) | (df["Assault"] < 0) | (df["UrbanPop"] < 0)]

,Unnamed: 0,Murder,Assault,UrbanPop


In [7]:
df.isna().sum()

Unnamed: 0    0
Murder        0
Assault       1
UrbanPop      0
dtype: int64

## Flagged Values: Comments  and Actions

For Georgia: "Assault = NaN." This value is missing. It should not be deleted or replaced unless there is enough information to justify the change. I would leave it missing and document it.

For Iowa: "UrbanPop = 570." This value is mathematically impossible because an urban population percentage cannot be greater than 100%. I would flag it as an error and investigate the original data before changing it.

For South Carolina: "Assault = 879." In comparison with the other states, this value is extremely high. However, I prefer to keep it flagged and investigate the original data before deciding whether it should be corrected or removed.

In [8]:
df.columns

Index(['Unnamed: 0', 'Murder', 'Assault', 'UrbanPop'], dtype='str')